05-setting-up.Rmd

```diff
@@ -9,35 +9,163 @@ ottrpal::set_knitr_image_path()
+- [A BioStars thread where many individuals share their own organizational strategies](https://www.biostars.org/p/821/) [@Biostars2021].
+- [Data Carpentry course chapter about getting organized](https://bioinformatics-core-shared-training.github.io/shell-genomics/07-organization/index.html) [@DataCarpentry2019].
 
 ## Navigate file paths
 
+In point and click apps (called Graphics User Interfaces) you navigate to files by clicking on folders. But for R programming and other command line interfaces, we navigate and use files by using `file paths`. `File paths` are series of folders it takes to get to the file, not unlike a street address.
+
+To make an analogy, if someone asked you directions to a particular building, the directions you give would be tailored based on where this person located. In other words your directions would be relative to their location.
+
+But file paths can be *relative* or *absolute*.
+
+In the same way, your computer can be given absolute directions to a file - basically the directions with absolute directions or they can be relative to where you are calling the command in the computer.
+
 ```{r, fig.align='center', out.width="100%", echo = FALSE, fig.alt= "Major point!! example image"}
 ottrpal::include_slide("https://docs.google.com/presentation/d/1MNHf8JpolaEP_vQ_kB-1xRBF9wo3haCArRu117hBoHA/edit#slide=id.g2fea8805c08_0_1337")
 ```
+So in our above analogy, if you are trying to direct someone to somewhere on Johns Hopkins’ campus with a file path:
+
+An absolute file path would be:
+`/Earth/North America/United States/Maryland/Baltimore/Johns Hopkins University/Street Name/Building number`
+
+Whereas if the person was already in Baltimore, a relative file path would be:
+`Johns Hopkins University/Street Name/Building number`
+
+The end of a path string may be a file name if you are creating a path to a file. If you are creating a path to a folder, the path string will end with the destination folder.
+
+To know your location within a file system is to know exactly what folder you are in right now. The folder that you are in right now is called the `working directory` aka your "Current Location".  In the above analogy a person being located in Baltimore would be their working directory. In a path, folder names are separated by forward slashes `/`
+
+Note that a relative directory may be different between different apps: RStudio versus Terminal versus something else. So you if you switch between the `Console` and `Terminal` tabs, you will have to pay attention to what your `working directory` is. This is also different from the `Files` pane which has no bearing on your working directory either.
+
+Returning to computer files. In your Terminal you can see your working directory at the top of the Terminal window or at the beginning of the terminal prompt. Knowing this, this can tell you how you need to change the command you are entering. Let’s say you want to list, using the `ls` command, a file called `file.txt`.
```

Comment:
maybe need an image of this

Reply:
I started one here but had to leave haha: https://docs.google.com/presentation/d/1MNHf8JpolaEP_vQ_kB-1xRBF9wo3haCArRu117hBoHA/edit#slide=id.g31fc7298e99_0_0

Reply:
ok Im adding one... but I feel like we need an exercise with some command line stuff

Reply:
That might be its own chapter before we get here?

Reply:
yeah and file paths are so hard for people to understand too - maybe better on it's own as well?

Reply:
want to merge this and then decide if we want to expand later?